DatadogRUM/Sources/Integrations/WebViewEventReceiver.swift

```diff
@@ -63,47 +63,55 @@ internal final class WebViewEventReceiver: FeatureMessageReceiver {
     /// - Parameters:
     ///   - event: The Browser RUM event.
     ///   - core: The core to write the event.
-    func write(event: JSON, to core: DatadogCoreProtocol) {
+    private func write(event: JSON, to core: DatadogCoreProtocol) {
         commandSubscriber.process(
             command: RUMKeepSessionAliveCommand(
                 time: dateProvider.now,
                 attributes: [:]
             )
         )
 
-        core.scope(for: RUMFeature.name)?.eventWriteContext { context, writer in
-            guard
-                let rum: RUMCoreContext = try? context.baggages[RUMFeature.name]?.decode()
-            else {
-                return writer.write(value: AnyEncodable(event))
+        core.scope(for: RUMFeature.name)?.eventWriteContext { [weak core] context, writer in
+            guard let rumBaggage = context.baggages[RUMFeature.name] else {
+                return // RUM not enabled
             }
 
-            var event = event
-
-            if let date = event["date"] as? Int {
-                let viewID = (event["view"] as? JSON)?["id"] as? String
-                let serverTimeOffsetInMs = self.getOffsetInMs(viewID: viewID, context: context)
-                let correctedDate = Int64(date) + serverTimeOffsetInMs
-                event["date"] = correctedDate
-            }
-
-            if var application = event["application"] as? JSON {
-                application["id"] = rum.applicationID
-                event["application"] = application
-            }
-
-            if var session = event["session"] as? JSON {
-                session["id"] = rum.sessionID
-                event["session"] = session
-            }
-
-            if var dd = event["_dd"] as? JSON, var dd_sesion = dd["session"] as? [String: Int] {
-                dd_sesion["plan"] = 1
-                dd["session"] = dd_sesion
-                event["_dd"] = dd
+            do {
+                let rumContextOptional: RUMCoreContext? = try rumBaggage.decode()
+
+                guard let rum = rumContextOptional else {
+                    return // RUM session not sampled
+                }
```

Comment:
Not super cool, but real as we want to have both `nil` comparison and error handling. Related to [RUM-1197](https://datadoghq.atlassian.net/browse/RUM-1197).

Reply:
We have a fix in #1504, we might want to merge it first? we will have conflicts in WVT tests :/

Reply:
I'll wait for #1504

Reply:
Rebased 👌